### PR TITLE
Stabilize updateGitConfig tests

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
v0.12.0 (#917) can't be released yet because tests are failing on GitLab: [gitlab.com/iterative.ai/cml→2213159659:152](https://gitlab.com/iterative.ai/cml/-/jobs/2213159659#L152)

Template tests resort to a nondeterministic value (the `GITLAB_USER_NAME` environment variable) that is `undefined` when running the tests on GitHub, but becomes `Dmitry Petrov` when running the tests on GitLab.

This pull request hardcodes the authentication user name to `token` because this value is ignored by GitHub and GitLab. As long as it's not an empty string, it will work.

_Note: we didn't notice when the change was made because GitLab tests only run on the main branch after https://github.com/iterative/itops/issues/111._

<details>

That `undefined` from `this.userName` is replaced by `john` (the argument to the `userName` parameter of the `updateGitConfig` function) used during tests, due to a [short–circuit evaluation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR#short-circuit_evaluation).

In other providers, the `userName` function is deterministic, so this error doesn't happen.

### GitHub
https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/github.test.js#L46

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/github.js#L672-L674

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/github.js#L644

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/github.js#L650

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/github.test.js#L52

### GitLab
https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/gitlab.test.js#L51-L54

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/gitlab.js#L452-L454

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/gitlab.js#L428

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/gitlab.js#L433

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/gitlab.test.js#L59

### Bitbucket
https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/bitbucket_cloud.test.js#L48-L51

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/bitbucket_cloud.js#L288

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/bitbucket_cloud.js#L255

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/bitbucket_cloud.js#L260

https://github.com/iterative/cml/blob/15c325c30eda5571fea37bee8415151f8b414e2e/src/drivers/bitbucket_cloud.js#L271

</details>

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
> _It feels like we have to separate the authentication user name from the commit author user name. :thinking_face: Users may want to change the latter (aesthetic/attribution purposes), but never the former (get a gratuitous 403 error)._

File: src/drivers/gitlab.test.js [hunk 1]
```diff
@@ -54,9 +54,9 @@ describe('Non Enviromental tests', () => {
     });
     expect(command).toMatchInlineSnapshot(`
       "
-          git config user.name \\"john\\" &&
+          git config user.name \\"token\\" &&
```
Comment (0x2b3bfa0) [inline src/drivers/gitlab.test.js:57]:
```suggestion
          git config user.name \\"john\\" &&
```

Whoops!

**Review by casperdcl — APPROVED** [review]
lgtm but just to double-check... @DavidGOrtega do you see any potential problems with this change?

**Comment by DavidGOrtega** [comment]
As @0x2b3bfa0 says its OK as its not an empty string. At least for now.

**Pull request merged**